src/main/java/org/breedinginsight/services/ProgramService.java

```diff
@@ -118,6 +119,25 @@ public Program create(ProgramRequest programRequest, AuthenticatedUser actingUse
             throw new UnprocessableEntityException("Species does not exist");
         }
 
+        //Check that key present
+        if (programRequest.getKey().isBlank()){
+            throw new UnprocessableEntityException("Program key required");
+        }
+
+        //Check that program key not already in use
+        if (programKeyInUse(programRequest.getKey())) {
+            throw new AlreadyExistsException("Program key already in use");
+        }
+
+        //Ensure program key uppercase
+        programRequest.setKey(programRequest.getKey().toUpperCase());
+
+        //Check that program key formatting correct
+        ArrayList<String> keyErrors = dao.getKeyValidationErrors(programRequest.getKey());
```

Comment:
Code style change -> Can you update the type for the `keyErrors` variable to be `List<String>`.  `List` is an interface, where `ArrayList` is an implementation of `List`.  It's best practice to use the interface as the defined type.

Reply:
Fixed.